### PR TITLE
Avoid sharing type in typeconstraint

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java [hunk 1]
```diff
@@ -671,7 +670,7 @@ void executeInstruction(Instruction instruction) {
         pop = popStack(1, instruction.opcode);
         sv = pop.values.get(0);
         TypedConstraint typedConstraint = programState.getConstraint(sv, TypedConstraint.class);
-        Type type = typedConstraint != null ? typedConstraint.type : Symbols.unknownType;
+        Type type = typedConstraint != null ? typedConstraint.getType(semanticModel) : Symbols.unknownType;
```
Comment (saberduck) [inline java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java]: this means we will still keep reference to `Type` inside exceptional symbolic value, which will be stored in yield and thus shared in the behavior cache. It seems to me that we need to do same change in `ExceptionalYield`

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BytecodeEGWalker.java#L767](https://github.com/SonarSource/sonar-java/blob/e88538bf2bd4b7f33d28447b03996b10be4d7a4a/java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java#L767): Take the required action to fix the issue indicated by this comment. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BytecodeEGWalker.java#L767](https://github.com/SonarSource/sonar-java/blob/8ca63f0d3405d90f15f8d87f52d90a4d6666c3d0/java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java#L767): Take the required action to fix the issue indicated by this comment. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java (hunk 1 above)
Comment (benzonico) [reply java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java]: see second commit.

**Pull request merged**